src/components/views/messages/MessageEvent.tsx

```diff
@@ -64,7 +63,9 @@ export interface IOperableEventTile {
     getEventTileOps(): IEventTileOps | null;
 }
 
-const baseBodyTypes = new Map<string, typeof React.Component>([
+type ComponentType = React.ComponentType<IBodyProps & React.RefAttributes<any>>;
```

Comment:
It's not obvious to me that `React.RefAttributes` is the right thing here. (In particular, it inherits from `React.Attributes`, which seems to add a bunch of stuff that we don't want).

Shouldn't we just declare `ref?: Ref<any>` explicitly?

Reply:
It actually feels like it would be nicer to define a `BodyPropsWithRef` type next to `IBodyProps`, or something, so that this can just become `React.ComponentType<BodyPropsWithRef>`, and then we can actually document the difference between `BodyPropsWithRef` and `IBodyProps` and when you want one or the other. At the moment you have to do a lot of staring at code to figure it out.